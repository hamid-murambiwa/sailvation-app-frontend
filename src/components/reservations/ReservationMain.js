import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchAllDestinations } from '../../redux/Destinations/destinations';
import { Link } from 'react-router-dom';

function ReservationMain() {

  const dispatch = useDispatch();
  // const userId = 1;
  useEffect(() => {
    dispatch(fetchAllDestinations());
  }, []);

  const destinations = useSelector((state) => state.destinationReducer);
  console.log(destinations.data);

  return (
    <div>
      <h1>Available destinations</h1>
      <ul className="cruise-list">
      { destinations.data && destinations.data.map((destination) => (
      <li key={destination.id} className="reservation-list-item">
      <div>
        <h2>{destination.city}</h2>
        <p>{destination.country}</p>
        <p>{destination.departure_city}</p>
        <p>{destination.days}</p>
        <p>{destination.price}</p>
        <img src={destination.image} alt={'an image of ' + destination.city} />
      </div>
      {/* <Link to={`/cruises/${reservation.cruise_id}/reservation/${reservation.id}/destination`}>
        Reservation
      </Link> */}
    </li>
    ))}
      </ul>
    </div>
  );
}

export default ReservationMain;
