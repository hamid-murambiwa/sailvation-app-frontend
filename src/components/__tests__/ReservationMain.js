import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom/cjs/react-router-dom.min';
import { Provider } from 'react-redux';
import store from '../__mocks__/configureStore';
import ReservationMain from '../reservations/ReservationMain';

describe('Login page is working fine', () => {
  test('renders Login', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <ReservationMain />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
