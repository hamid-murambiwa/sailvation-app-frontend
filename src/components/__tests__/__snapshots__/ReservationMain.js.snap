// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page is working fine renders Login 1`] = `
<h1
  className="text-center"
>
   You have to Log in
</h1>
`;
